react/features/toolbox/components/web/EndConferenceButton.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+import { endConference } from '../../../base/conference';
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import { AbstractButton, type AbstractButtonProps } from '../../../base/toolbox/components';
+
+/**
+ * The type of the React {@code Component} props of {@link EndConferenceButton}.
+ */
+type Props = AbstractButtonProps & {
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function
+};
+
+/**
+ * An implementation of a button for ending the meeting.
+ */
+class EndConferenceButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.endConference';
+    label = 'toolbar.endConference';
+    tooltip = 'toolbar.endConference';
+
+    /**
+     * Handles clicking / pressing the button.
+     *
+     * @override
+     * @protected
+     * @returns {void}
+     */
+    _handleClick() {
+
+        const { handleClick } = this.props;
+
+        if (handleClick) {
+            handleClick();
+
+            return;
+        }
+
+        this.props.dispatch(endConference());
+    }
+
+}
+
+/**
+ * Maps part of the redux state to the component's props.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {Object} ownProps - The properties explicitly passed to the component instance.
+ * @returns {Props}
+ */
+function _mapStateToProps(state, ownProps) {
+    const { visible = true } = ownProps;
+
+    return {
+        hideIcon: true,
```

Comment:
Extract all of these to a constant object, since they don't change. Just leave the props that do change with the redux state here.